cmd/kubelet/app/server.go

```diff
@@ -380,6 +380,11 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 		return err
 	}
 
+	kubeClient, err := getKubeClient(s)
```

Comment:
Besides possibly making the next step a little easier, an advantage to piggybacking on the sync is that a misbehaving isolator in this impl could cause the Kubelet to spam the API server with node status updates.